### PR TITLE
APIM-490 feat: define template for log-v4 and request-v4

### PR DESCRIPTION
**Pull request by jgiovaresco**
**Issue**

https://gravitee.atlassian.net/browse/APIM-490

**Description**

**Additional context**

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `4.1.0-support-v4-logging-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/elasticsearch/gravitee-common-elasticsearch/4.1.0-support-v4-logging-SNAPSHOT/gravitee-common-elasticsearch-4.1.0-support-v4-logging-SNAPSHOT.zip)

### CODE REVIEW

**Review by guillaumelamirand — APPROVED** [review]

**Review by phiz71 — COMMENTED** [review]
Why is it required to add a dependency on reporter-api ?

**Comment by jgiovaresco** [comment]
> Why is it required to add a dependency on reporter-api ?

I suppose it is used for helping the autocompletion https://github.com/gravitee-io/gravitee-common-elasticsearch/pull/179/files#diff-0276d96c303e69d44719acc9022ec2d5e12ee72562c9a3966df3d9fcefe73b5eR3

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            "client-identifier": {
+                "type": "keyword"
+            },
+            "request-ended": {
+                "type": "boolean"
+            },
+            "entrypoint-request": {
+                "type": "object",
+                "properties": {
+                    "body":{
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:32]: The `uri` and `method` are missing. I'm wondering if we should add them to the template 🤔

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    "headers":  {
+                       "enabled":  false,
+                       "type": "object"
+                   }
+                }
+            },
+            "entrypoint-response": {
+                "type": "object",
+                "properties": {
+                    "body":{
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:44]: The `status` is missing. I'm wondering if we should add it to the template 🤔

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    "headers":  {
+                       "enabled":  false,
+                       "type": "object"
+                   }
+                }
+            },
+            "endpoint-response": {
+                "type": "object",
+                "properties": {
+                    "body":{
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:68]: The `status` is missing. I'm wondering if we should add it to the template 🤔

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    "headers":  {
+                       "enabled":  false,
+                       "type": "object"
+                   }
+                }
+            },
+            "endpoint-request": {
+                "type": "object",
+                "properties": {
+                    "body":{
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:56]: The `uri` and `method` are missing. I'm wondering if we should add them to the template 🤔

File: src/main/resources/freemarker/es7x/mapping/index-template-metrics-v4.ftl [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+                "type": "integer",
+                "index": false
+            },
+            "request-ended": {
+                "type": "boolean"
+            },
+            "endpoint": {
+                "type": "keyword"
+            },
+            "endpoint-response-time": {
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-metrics-v4.ftl:122]:
```suggestion
            "endpoint-response-time-ms": {
```

File: src/main/resources/freemarker/es7x/mapping/index-template-metrics-v4.ftl [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+                "type": "long"
+            },
+            "status": {
+                "type": "short"
+            },
+            "response-content-length": {
+                "type": "integer",
+                "index": false
+            },
+            "gateway-latency": {
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-metrics-v4.ftl:132]:
```suggestion
            "gateway-latency-ms": {
```

File: src/main/resources/freemarker/es7x/mapping/index-template-metrics-v4.ftl [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+                "type": "integer"
+            },
+            "gateway-response-time": {
```
Comment (jgiovaresco) [inline src/main/resources/freemarker/es7x/mapping/index-template-metrics-v4.ftl:135]:
```suggestion
            "gateway-response-time-ms": {
```

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl (hunk 1 above)
Comment (guillaumelamirand) [reply src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:32]: Don't know, not present in v2 so I just did the same thing.

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl (hunk 2 above)
Comment (guillaumelamirand) [reply src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:44]: Don't know, not present in v2 so I just did the same thing.

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl (hunk 4 above)
Comment (guillaumelamirand) [reply src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:56]: Don't know, not present in v2 so I just did the same thing.

File: src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl (hunk 3 above)
Comment (guillaumelamirand) [reply src/main/resources/freemarker/es7x/mapping/index-template-log-v4.ftl:68]: Don't know, not present in v2 so I just did the same thing.

**Comment by guillaumelamirand** [comment]
> Why is it required to add a dependency on reporter-api ?

@jgiovaresco answered you. It is to fil the template with the correct type.

**Pull request merged**